### ISSUE TITLE
ruby 3.0.0: undefined method `arity' for

### ISSUE THREAD

**Issue by clwy-cn** [collectiveidea/awesome_nested_set#439]
This error is only in ruby 3.0.0
if I use ruby 2.6.6 or 2.7.2，it works fine.

My Rails version is 6.1.0

This code is causing the problem：

```
acts_as_nested_set
```

undefined method `arity' for {:class_name=>"Guide::Page", :foreign_key=>"parent_id", :primary_key=>"id", :counter_cache=>:children_count, :inverse_of=>:children, :polymorphic=>false, :touch=>false, :optional=>true}:Hash

![image](https://user-images.githubusercontent.com/57120683/103219706-6f1caf80-4959-11eb-910b-a812fdb85659.png)

**Comment by parndt** [comment]
Does it work properly if you use the repo version?

```ruby
gem 'awesome_nested_set', github: 'collectiveidea/awesome_nested_at', branch: 'master'
```

**Comment by parndt** [comment]
Alternatively, those changes are now available using version 3.3.0:

```ruby
gem 'awesome_nested_set', '~> 3.3.0'
```

**Comment by damianlegawiec** [comment]
This is fixed with 3.3.x , thank you @parndt 🙏

**Comment by parndt** [comment]
Thanks to @krisdigital 🙇
